spec/lib/response-handler.spec.js

```diff
@@ -0,0 +1,128 @@
+"use strict";
+
+describe("lib/response-handler", () => {
+    var bluebird, error, logger, parseLinkHeader, responseHandler, URI;
+
+    bluebird = require("bluebird");
+    error = require("../../lib/error")();
+    logger = require("../../lib/logger")();
+    parseLinkHeader = require("parse-link-header");
+    URI = require("urijs");
+    responseHandler = require("../../lib/response-handler")(bluebird, error, logger, parseLinkHeader, URI);
+
+    describe(".isErrorCode()", () => {
+        it("returns false if the status code is within the right range", () => {
+            expect(responseHandler.isErrorCode(300)).toBe(false);
+        });
+        it("returns true if the status code is less than 200", () => {
+            expect(responseHandler.isErrorCode(150)).toBe(true);
+        });
+        it("returns true if the status code is greater than 399", () => {
+            expect(responseHandler.isErrorCode(404)).toBe(true);
+        });
+    });
+    describe(".createErrorForResponse()", () => {
+        var errorResponse;
+
+        it("sets the resposne", () => {
+            errorResponse = {
+                error: "ExampleError",
+                response: {
+                    body: {
+                        code: "duplicate_artifact",
+                        message: "Example Error Message"
+                    }
+                }
+            };
+            expect(responseHandler.createErrorForResponse(errorResponse)).toEqual(new Error("Example Error Message"));
+        });
+        it("returns any unkown errors", () => {
+            errorResponse = {
+                body: {
+                    code: "anUnkownError"
+                }
+            };
+            expect(responseHandler.createErrorForResponse(errorResponse)).toEqual(new Error("UNKOWN", "UNKOWN"));
```

Comment:
This assertion doesn't appear to actually test anything.

```
-            expect(responseHandler.createErrorForResponse(errorResponse)).toEqual(new Error("UNKOWN", "UNKOWN"));
+            expect(responseHandler.createErrorForResponse(errorResponse)).toEqual(new Error("NOT_UNKOWN", "ITS A VERY KNOWN ERROR!"));
```

Especially since it should be "UNKNOWN" not "UNKOWN"

This goes for all the other error testing in this test. You may want to invest in an `assertError` function